keras/callbacks.py

```diff
@@ -807,9 +807,9 @@ def on_train_end(self, _):
     def on_batch_end(self, batch, logs=None):
         logs = logs or {}
 
-        if self.write_batch_performance == True:
+        if self.write_batch_performance:
```

Comment:
It's better use a flag to control write epoch-level summary or batch-level summary. The batch-level summary will be overwrite in your implementation.